src/style.rs

```diff
@@ -603,6 +603,195 @@ impl Protection {
+    pub height: f64,
+    /// Whether the height is custom (manually set)
+    pub custom_height: bool,
+    /// Whether the row is hidden
+    pub hidden: bool,
+    /// Thick top border
+    pub thick_top: bool,
+    /// Thick bottom border
+    pub thick_bottom: bool,
+}
+
+impl RowHeight {
+    /// Create a new row height
+    pub fn new(row: u32, height: f64) -> Self {
+        Self {
+            row,
+            height,
+            custom_height: false,
+            hidden: false,
+            thick_top: false,
+            thick_bottom: false,
+        }
+    }
+
+    /// Set custom height flag
+    pub fn with_custom_height(mut self, custom: bool) -> Self {
+        self.custom_height = custom;
+        self
+    }
+
+    /// Set hidden flag
+    pub fn with_hidden(mut self, hidden: bool) -> Self {
+        self.hidden = hidden;
+        self
+    }
+
+    /// Set thick top border
+    pub fn with_thick_top(mut self, thick_top: bool) -> Self {
+        self.thick_top = thick_top;
+        self
+    }
+
+    /// Set thick bottom border
+    pub fn with_thick_bottom(mut self, thick_bottom: bool) -> Self {
+        self.thick_bottom = thick_bottom;
+        self
+    }
+
+    /// Check if row is visible
+    pub fn is_visible(&self) -> bool {
+        !self.hidden
+    }
+}
+
+/// Worksheet layout information
+#[derive(Debug, Clone, Default, PartialEq)]
+pub struct WorksheetLayout {
+    /// Column widths
+    pub column_widths: Vec<ColumnWidth>,
+    /// Row heights
+    pub row_heights: Vec<RowHeight>,
+    /// Default column width
+    pub default_column_width: Option<f64>,
+    /// Default row height
+    pub default_row_height: Option<f64>,
+}
+
+impl WorksheetLayout {
+    /// Create a new worksheet layout
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Add a column width
+    pub fn add_column_width(mut self, column_width: ColumnWidth) -> Self {
+        self.column_widths.push(column_width);
+        self
+    }
+
+    /// Add a row height
+    pub fn add_row_height(mut self, row_height: RowHeight) -> Self {
+        self.row_heights.push(row_height);
+        self
+    }
+
+    /// Set default column width
+    pub fn with_default_column_width(mut self, width: f64) -> Self {
+        self.default_column_width = Some(width);
+        self
+    }
+
+    /// Set default row height
+    pub fn with_default_row_height(mut self, height: f64) -> Self {
+        self.default_row_height = Some(height);
+        self
+    }
+
+    /// Get column width for a specific column
+    pub fn get_column_width(&self, column: u32) -> Option<&ColumnWidth> {
+        self.column_widths.iter().find(|cw| cw.column == column)
+    }
+
+    /// Get row height for a specific row
+    pub fn get_row_height(&self, row: u32) -> Option<&RowHeight> {
+        self.row_heights.iter().find(|rh| rh.row == row)
+    }
+
+    /// Get effective column width (custom or default)
+    pub fn get_effective_column_width(&self, column: u32) -> f64 {
```

Comment:
This effective width is probably not the width that the user would expect and also may be dependent on the default font for the file.

See the following for the calculation in the other direction: https://github.com/jmcnamara/rust_xlsxwriter/blob/main/src/worksheet.rs#L19340-L19368

However, there may not be any simple way to return the correct value in all cases.